### PR TITLE
add -no-prefix cli argument

### LINKED ISSUE #198: Provide "no_prefix" option from the command line
There is a way to provide `no_prefix` option from the command line as an argument? I did not find an equivalent option by `envconsul --help`

### PR DESCRIPTION
**Pull request by eikenb**
Allows speficying no-prefix globally for all prefixes/secrets from the
command line. Argument handling must be done after standard parsing as
it relies on -prefix and -secret already being set.

Fixes #198

### CODE REVIEW

File: cli.go [hunk 1]
```diff
@@ -614,6 +622,16 @@ func (cli *CLI) ParseFlags(args []string) (*Config, []string, bool, bool, error)
 		return nil, nil, false, false, err
 	}
 
+	// Post-processing of no-prefix option
+	if no_prefix != nil {
+		for _, p := range *c.Prefixes {
+			p.NoPrefix = no_prefix
```
Comment (kyhavlov) [inline cli.go]: We should probably only set this value if NoPrefix is nil for the prefix, that way it can still act as an override.  Same goes for the secrets below.

File: cli.go [hunk 2]
```diff
@@ -768,6 +786,12 @@ Options:
       Set the maximum staleness and allow stale queries to Consul which will
       distribute work among all servers instead of just the leader
 
+  -no-prefix[=<bool>]
+	  Tells Envconsul to not prefix the keys with their parent "folder".
+	  Default for -prefix is true, default for -secret is false.
+	  Differing defaults is to maintain backward compatibility.
+	  Leaving off the bool value is the same as setting to true.
```
Comment (kyhavlov) [inline cli.go]: I think we can document this flag as `-no-prefix` instead of `-no-prefix[=<bool>]` like we do with `-once` and `-sanitize`. That's simpler and then we don't need this last line either.

File: cli.go (hunk 1 above)
Comment (eikenb) [reply cli.go]: I think the only time that (NoPrefix not being nil) would come up is if they had the NoPrefix set via the config file. Which would mean that the config file settings would override the command line argument. Isn't it usually the opposite, with the command line arguments overriding the config file?

File: cli.go (hunk 2 above)
Comment (eikenb) [reply cli.go]: I mentioned the argument for -no-prefix as, unlike the other cases, both forms are useful as each of the 2 prefix systems controlled (the consul keys and vault's secrets) have opposite defaults.

**Comment by eikenb** [comment]
@kyhavlov Thanks for the review!
Responses below.

File: cli.go (hunk 2 above)
Comment (eikenb) [reply cli.go]: I do think that is a bit wordy for cli reference output. It should only contain what it does. I think adding the other information to the documented config file in the Readme would be a better place for it.

File: cli.go (hunk 1 above)
Comment (eikenb) [reply cli.go]: I dug around just to be sure and the command line argument taking precedence is definitely the de-facto cross platform standard. So I think we should keep it as is.

Comment (pearkes) [reply cli.go]: Giving the CLI precedence makes sense to me 👍
